### PR TITLE
Improve some exception types

### PR DESCRIPTION
**Pull request by RoosterDragon**
Throw more specific subclasses of `Exception`.

### CODE REVIEW

**Comment by Phrohdoh** [comment]
I don't see why not. 👍

File: OpenRA.Mods.Common/Traits/Multipliers/UpgradableMultiplierTrait.cs [hunk 1]
```diff
@@ -45,7 +45,7 @@ public abstract class UpgradeMultiplierTrait : IUpgradable, IDisabledTrait, ISyn
 		protected UpgradeMultiplierTrait(UpgradeMultiplierTraitInfo info, string modifierType, string actorType)
 		{
 			if (info.Modifier.Length == 0)
-				throw new Exception("No modifiers in " + modifierType + " for " + actorType);
+				throw new ArgumentException("No modifiers in " + modifierType + " for " + actorType);
```
Comment (pchote) [inline OpenRA.Mods.Common/Traits/Multipliers/UpgradableMultiplierTrait.cs]: Missing the argument name here?

Comment (RoosterDragon) [inline OpenRA.Mods.Common/Traits/Multipliers/UpgradableMultiplierTrait.cs]: Any of the three arguments could need changing here if you see this exception - I usually only specify an argument if it is the only one that could affect the check.

Comment (pchote) [inline OpenRA.Mods.Common/Traits/Multipliers/UpgradableMultiplierTrait.cs]: Ah right, I see.

**Review by pchote — APPROVED** [review]
These are all obviously more correct, so :+1:

**Pull request merged**